common/src/cellular_at_core.c

```diff
@@ -56,49 +56,38 @@ typedef enum CellularATStringValidationResult
 
 /*-----------------------------------------------------------*/
 
-static CellularATError_t validateString( const char * pString,
-                                         CellularATStringValidationResult_t * pStringValidationResult );
+static void validateString( const char * pString,
+                            CellularATStringValidationResult_t * pStringValidationResult );
 static uint8_t _charToNibble( char c );
 
 /*-----------------------------------------------------------*/
 
-static CellularATError_t validateString( const char * pString,
-                                         CellularATStringValidationResult_t * pStringValidationResult )
+static void validateString( const char * pString,
+                            CellularATStringValidationResult_t * pStringValidationResult )
 {
-    CellularATError_t atStatus = CELLULAR_AT_SUCCESS;
     size_t stringLength = 0U;
 
-    if( ( pString == NULL ) || ( pStringValidationResult == NULL ) )
+    /* The strnlen() function returns strlen(s), if that is less than maxlen,
+        * or maxlen if there is no null terminating ('\0') among the first
+        * maxlen characters pointed to by s.
+        *
+        * stringLength == CELLULAR_AT_MAX_STRING_SIZE is valid because it means that
+        * ( CELLULAR_AT_MAX_STRING_SIZE + 1 ) character is null terminating
+        * character.*/
```

Comment:
```suggestion
    /* The strnlen() function returns strlen(s), if that is less than maxlen,
     * or maxlen if there is no null terminating ('\0') among the first
     * maxlen characters pointed to by s.
     *
     * stringLength == CELLULAR_AT_MAX_STRING_SIZE is valid because it means that
     * ( CELLULAR_AT_MAX_STRING_SIZE + 1 ) character is null terminating
     * character.*/
```